docs/API/Instruction.md

```diff
@@ -59,52 +45,128 @@ public PlayerCondition parsePlayer(final Instruction instruction) throws QuestEx
 Compare [itemdurability](../Documentation/Scripting/Building-Blocks/Events-List.md#item-durability-itemdurability) event.
 ```JAVA title="ItemDurabilityEventFactory.java"
 public PlayerEvent parsePlayer(final Instruction instruction) throws QuestException {
-    final Variable<EquipmentSlot> slot = instruction.get(instruction.getParsers().forEnum(EquipmentSlot.class));
-    final Variable<PointType> operation = instruction.get(instruction.getParsers().forEnum(PointType.class));
-    final Variable<Number> amount = instruction.get(instruction.getParsers().number());
+    final Variable<EquipmentSlot> slot = instruction.enumeration(EquipmentSlot.class).get();
+    final Variable<PointType> operation = instruction.enumeration(PointType.class).get();
+    final Variable<Number> amount = instruction.number().get();
     final boolean ignoreUnbreakable = instruction.hasArgument("ignoreUnbreakable");
     final boolean ignoreEvents = instruction.hasArgument("ignoreEvents");
     
     //creating event object and returning it...
 }
 ```
 
+Compare [delay](../Documentation/Scripting/Building-Blocks/Objectives-List.md#wait-delay) objective.
+```java title="DelayObjectiveFactory.java"
+public Objective parseInstruction(final Instruction instruction) throws QuestException {
+    final Variable<Number> delay = instruction.number().atLeast(0).get();
+    final Variable<Number> interval = instruction.number().atLeast(1).get("interval", 20 * 10);
+    //creating objective object and returning it...
+}
+```
+
+### Argument Parsing
+The argument parsing step represents the method of converting a string into a java type is decided.
+The instruction chain may be accessed conveniently, starting directly from any `Instruction` instance.
+
+| Call | Description |
+| :--- | :---------- |
+ | `instruction.number()` | Default parser for `java.lang.Number` covering both integer and floating point values |
+ | `instruction.string()` | Default parser for `java.lang.String` |
+ | `instruction.bool()` | Default parser for `java.lang.Boolean` |
+ | `instruction.uuid()` | Default parser for `java.util.UUID` |
+ | `instruction.world()` | Default parser for `org.bukkit.World` |
+ | `instruction.location()` | Default parser for `org.bukkit.Location` |
+ | `instruction.vector()` | Default parser for `org.bukkit.util.Vector` |
+ | `instruction.component()` | Default parser for `net.kyori.adventure.text.Component` |
+ | `instruction.item()` | Default parser for `org.betonquest.betonquest.api.instruction.type.ItemWrapper` representing items defined in BetonQuest |
+ | `instruction.packageIdentifier()` | Default parser for package identifiers producing a `java.lang.String` |
```

Comment:
Creates a identifier from a String, where a string that already contains a package is untouched, a path to a package is not resolved, if no package is present, the package is added